ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -297,11 +298,30 @@ private void handleQuery(final RequestContext info, final Query query,
           return;
         }
         case KSTREAM: {
-          throw new KsqlStatementException(
-              "Pull queries are not supported on streams."
-                  + PullQueryValidator.PULL_QUERY_SYNTAX_HELP,
-              statement.getStatementText()
+          final StreamPullQueryMetadata queryMetadata =
+              ksqlEngine.createStreamPullQuery(
+                  info.securityContext.getServiceContext(),
+                  analysis,
+                  configured,
+                  true
+              );
+
+          localCommands.ifPresent(lc -> lc.write(queryMetadata.getTransientQueryMetadata()));
+
+          final PushQuerySubscription subscription =
+              new PushQuerySubscription(exec,
+                  streamSubscriber,
+                  queryMetadata.getTransientQueryMetadata()
+              );
+
+          log.info(
+              "Running query {}",
+              queryMetadata.getTransientQueryMetadata().getQueryId().toString()
           );
+          queryMetadata.getTransientQueryMetadata().start();
+
+          streamSubscriber.onSubscribe(subscription);
+          return;
```

Comment:
I have been looking in the code and it seems we don't have a `PushQueryPublisherTest` like we do for `PullQueryPublisherTest`. Shouldn't we have one?

Reply:
Yeah, I noticed that, too. We could add one, but I didn't bother because the publisher is already covered by other tests, and the pull query publisher test looked of dubious value to me.